compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -3877,16 +3877,7 @@ object Parsers {
         val mods1 = addFlag(mods, Method)
         val ident = termIdent()
         var name = ident.name.asTermName
-        val paramss =
-          if in.featureEnabled(Feature.clauseInterleaving) then
```

Comment:
shouldn't this still be disabled when source version is below 3.6?

Reply:
It should yes

Reply:
@bishabosha I've also added a test to check the correct behavior with `-source 3.5`

Reply:
nice